RiotSwiftUI/Modules/Room/PollHistory/View/SegmentedPicker.swift

```diff
@@ -0,0 +1,61 @@
+// 
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct SegmentedPicker<Tag: Hashable>: View {
+    private let segments: [(String, Tag)]
+    private let selection: Binding<Tag>
+    private let interSegmentSpacing: CGFloat
+    
+    @Environment(\.theme) private var theme
+    
+    init(segments: [(String, Tag)], selection: Binding<Tag>, interSegmentSpacing: CGFloat) {
```

Comment:
Could we use an enum here rather than a tuple? I think that would be tidier in the long run.

Reply:
Yeah, I could have done better!
Now I fell better: https://github.com/vector-im/element-ios/pull/7267/commits/b4c8c0e63279f6733c6317d0d0e67f3b4afedcf6
😀